src/IceRpc.Retry/RetryInterceptor.cs

```diff
@@ -53,21 +54,22 @@ public class RetryInterceptor : IInvoker
     public RetryInterceptor(IInvoker next, RetryOptions options, ILogger logger)
     {
         _next = next;
-        _options = options;
+        _maxAttempts = options.MaxAttempts;
+        _maxPayloadSize = options.MaxPayloadSize;
         _logger = logger;
     }
 
     /// <inheritdoc/>
     public async Task<IncomingResponse> InvokeAsync(OutgoingRequest request, CancellationToken cancellationToken)
     {
-        if (request.PayloadContinuation is not null)
+        // This interceptor does not support retrying requests with a payload continuation.
+        if (request.PayloadContinuation is not null || _maxAttempts == 1)
```

Comment:
If max attempts is 1, there is no retrying and this interceptor shouldn't install a decorator.